File: demo/src/main/java/com/bytedance/scenedemo/navigation/reuse/ReuseDemoScene.java
package com.bytedance.scenedemo.navigation.reuse;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.bytedance.scene.group.GroupScene;
import com.bytedance.scenedemo.R;
import com.bytedance.scenedemo.utility.ColorUtil;

/**
 * Created by JiangQi on 8/13/18.
 */
public class ReuseDemoScene extends GroupScene {

    @NonNull
    @Override
    public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return (ViewGroup) inflater.inflate(R.layout.basic_layout, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        getView().setBackgroundColor(ColorUtil.getMaterialColor(getResources(), 0));

        TextView name = getView().findViewById(R.id.name);
        name.setText(R.string.nav_reuse_tip);

        Button btn = getView().findViewById(R.id.btn);
        btn.setText(R.string.nav_reuse_btn);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getNavigationScene().push(ReuseScene1.class);
            }
        });
    }
}

File: demo/src/main/java/com/bytedance/scenedemo/navigation/reuse/ReuseDemoScene.kt
package com.bytedance.scenedemo.navigation.reuse

import com.bytedance.scene.group.GroupScene
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.widget.Button
import com.bytedance.scenedemo.R
import android.widget.TextView
import com.bytedance.scenedemo.navigation.reuse.ReuseScene1
import com.bytedance.scenedemo.utility.ColorUtil

/**
 * Created by JiangQi on 8/13/18.
 */
class ReuseDemoScene : GroupScene() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        return inflater.inflate(R.layout.basic_layout, container, false) as ViewGroup
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view.setBackgroundColor(ColorUtil.getMaterialColor(resources, 0))
        val name = view.findViewById<TextView>(R.id.name)
        name.setText(R.string.nav_reuse_tip)
        val btn = view.findViewById<Button>(R.id.btn)
        btn.setText(R.string.nav_reuse_btn)
        btn.setOnClickListener { navigationScene!!.push(ReuseScene1::class.java) }
    }
}